pages/content/posts/complete-interop-features/index.md

```diff
@@ -0,0 +1,64 @@
+---
+author: "Jean-Yves Perrier"
+title: "Documenting missing interoperable web features"
+date: "2023-04-17"
+tags: ["projects"]
+ShowToc: false
+ShowBreadCrumbs: false
+excerpt: "Filling some MDN gaps by documenting missing interoperable features."
+---
+
+[MDN Web Docs](https://developer.mozilla.org/) is an invaluable set of information about the web platform. With about 12000 pages, it is the most extensive corpus of documents that web developers can refer to create their websites.
+
+Unfortunately, despite being almost 20 years old, many reference pages still need to be included. We call these missing pages _gaps_ and even have tools to detect some of them:
+
+- Mozilla's [Yari](https://github.com/mdn/yari/) renderer logs missing pages as flaws when it builds a page.
+- [Dominique Hazaël-Massieux](https://www.w3.org/People/Dom/) from the [W3C](https://www.w3.org/) created a tool ([mdn-gaps](https://dontcallmedom.github.io/mdn-gaps/)) listing all specification features without reference documentation. ![Extract of a screenshot listing a few random MDN gaps, as well as which browser supports them](mdn-gaps-example.png)
+
+These tools allow us to know the extent of the problem.
+
+## Why are there gaps on MDN?
+
+Gaps came organically. MDN Web Docs was, for many years, the home of a diverse set of documents: writers initially put much effort into documenting Firefox technologies like XUL and XBL, and time was scarce to focus on standardized languages. While CSS, HTML, and JavaScript had their documentation sets, docs for APIs were an afterthought. They were still living under a DOM hierarchy on MDN. With the advent of the HTML5 and CSS3 initiatives (and later the EcmaScript 6 one), Mozilla Developer Network's main subject shifted from proprietary to open and interoperable technologies. The shift was progressive, culminating in rebranding the site as _MDN Web Docs_, the archival of XUL and XBL docs, and, later, the move of Firefox Dev Tools docs to a dedicated [website](https://firefox-source-docs.mozilla.org/).
```

Comment:
```suggestion
Gaps came organically. MDN Web Docs is the home of a diverse set of documents. Initially, writers put great effort into documenting Firefox technologies like XUL and XBL, and time was scarce to focus on standardized languages. While CSS, HTML, and JavaScript were well documented, docs for APIs were an afterthought. They were still living under a DOM hierarchy on MDN. With the advent of HTML as a living standard and CSS module structure initiatives (and later the EcmaScript 6 one), Mozilla Developer Network's focus shifted from proprietary to open and interoperable technologies. The shift was progressive, culminating in rebranding the site as _MDN Web Docs_, the archiving of XUL and XBL docs, and, later, the move of [Firefox Dev Tools](https://firefox-source-docs.mozilla.org/) documentation to a dedicated website.
```

Reply:
There are still diverse docs on MDN. 
I live to avoid HTML5 and CSS3 as deprecated terms.